core/log/profiler_hook_summary.cpp

```diff
@@ -228,15 +265,40 @@ struct nested_summary {
 
     int64 overhead_ns{};
     std::vector<partial_entry> stack;
+    std::vector<time_point> free_list;
     std::unordered_map<std::pair<int64, int64>, int64, pair_hash> node_map;
     std::unordered_map<std::string, int64> name_map;
     std::vector<entry> nodes;
     std::vector<std::string> names;
+    std::shared_ptr<Timer> timer;
     bool broken{};
     bool check_nesting{};
     std::mutex mutex{};
 
-    nested_summary() { push("total"); }
+    nested_summary(std::shared_ptr<Timer> timer) : timer{std::move(timer)}
+    {
+        push("total");
+    }
+
+    time_point get_current_time_point()
```

Comment:
since this function and the one below are the same as in the non-nested summary, perhaps they should be extracted.

Reply:
done